helm/templates/deployment.yaml

```diff
@@ -54,28 +54,53 @@ spec:
         - name: {{ .Chart.Name }}
           securityContext:
             {{- toYaml .Values.securityContext | nindent 12 }}
-          # Conditional image selection
-          image: "{{ if .Values.ee.enabled }}{{ .Values.ee.image.repository }}{{ else }}{{ .Values.image.repository }}{{ end }}:{{ .Values.image.tag | default .Chart.Version }}"
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.Version }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
           workingDir: /app
           env:
             - name: PORT
               value: {{ .Values.service.port | quote }}
-            {{- if .Values.ee.enabled }}
+            {{- if .Values.management.enabled }}
             - name: MANAGEMENT_PORT
               value: {{ .Values.service.managementPort | quote }}
-            # EE-specific environment variables
+            - name: HUB_MODE
+              value: "database"
+            {{- if .Values.management.database.configMapName }}
```

Comment:
you shouldn't put another `if` here, if you enabled the management these values are required, you want to fail early not on the app